addons/payment/views/payment_form_templates.xml

```diff
@@ -329,7 +329,7 @@
             >
                 <t t-set="provider_id" t-value="provider_sudo.id"/>
             </t>
-            <div class="d-flex flex-column flex-md-row align-md-items-center justify-content-between
+            <div class="d-flex flex-column flex-sm-row align-md-items-center justify-content-between
                         gap-2 mt-2"
             >
```

Comment:
Missing a `flex-column` class to make it work in mobile. Also IMO `gap-3` is cleaner than `gap-2`, less crowded